extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -16,1552 +16,1606 @@
-        });
-
-        ctrl.canvas.on('selection:updated', function() {
-          onSelection();
-        });
-
-        ctrl.canvas.on('selection:cleared', function() {
-          ctrl.objectIsSelected = false;
-          ctrl.displayFontStyles = false;
-        });
-        $scope.$applyAsync();
-      };
-
-      ctrl.setCanvasDimensions = function() {
-        ctrl.canvas.setHeight(CANVAS_HEIGHT);
-        ctrl.canvas.setWidth(CANVAS_WIDTH);
-        ctrl.canvas.renderAll();
-      };
-
-      var initializeFabricJs = function() {
-        ctrl.canvas = new fabric.Canvas(ctrl.canvasID);
-        ctrl.setCanvasDimensions();
-        ctrl.canvas.selection = false;
-        ctrl.initializeMouseEvents();
-        createColorPicker('stroke');
-        createColorPicker('fill');
-        createColorPicker('bg');
-        // This is used to change the origin of shapes from top left corner
-        // to center of the shape. This is used to align the quadratic bezier
-        // control points correctly to the curve.
-        fabric.Object.prototype.originX = 'center';
-        fabric.Object.prototype.originY = 'center';
-      };
-
-      ctrl.$onInit = function() {
-        if (ctrl.value) {
-          ctrl.setSavedSvgFilename(ctrl.value, true);
-          var dimensions = (
-            ImagePreloaderService.getDimensionsOfImage(ctrl.value));
-          ctrl.svgContainerStyle = {
-            height: dimensions.height + 'px',
-            width: dimensions.width + 'px'
-          };
+        this.objectIsSelected = true;
+        var strokeWidthShapes = [
+          'rect', 'circle', 'path', 'line', 'polyline'];
+        if (strokeWidthShapes.indexOf(shape.get('type')) !== -1) {
+          this.fabricjsOptions.size = (
+            shape.get('strokeWidth').toString() + 'px');
+          this.displayFontStyles = false;
+        } else if (shape.get('type') === 'textbox') {
+          this.displayFontStyles = true;
+          this.fabricjsOptions.size = (
+            shape.get('fontSize' as keyof fabric.Object).toString() + 'px');
+          this.fabricjsOptions.fontFamily = shape.get(
+            'fontFamily' as keyof fabric.Object);
+          this.fabricjsOptions.italic = shape.get(
+            'fontStyle' as keyof fabric.Object) === 'italic';
+          this.fabricjsOptions.bold = shape.get(
+            'fontWeight' as keyof fabric.Object) === 'bold';
         } else {
-          angular.element(document).ready(function() {
-            initializeFabricJs();
-          });
+          this.displayFontStyles = false;
         }
-      };
-    }
-  ]
-});
+        // $scope.$applyAsync();
+      }
+    };
+
+    this.canvas.on('selection:created', () => {
+      onSelection();
+    });
+
+    this.canvas.on('selection:updated', () => {
+      onSelection();
+    });
+
+    this.canvas.on('selection:cleared', () => {
+      this.objectIsSelected = false;
+      this.displayFontStyles = false;
+    });
+    // $scope.$applyAsync();
```

Comment:
Should this be removed? Ditto in other places.

Reply:
Yup, removed in all other places.